tests_tf/test_attacks.py

```diff
@@ -250,6 +268,69 @@ def test_generate_np_clip_works_as_expected(self):
     self.assertClose(np.min(x_adv), -0.2)
     self.assertClose(np.max(x_adv), 0.1)
 
+class TestFastGradientMethod(CommonAttackProperties):
+
+  def setUp(self):
+    super(TestFastGradientMethod, self).setUp()
+
+    self.attack = FastGradientMethod(self.model, sess=self.sess)
+
+  def test_optimize_linear_l1(self):
+
+    # This test makes sure that `optimize_linear` actually finds the optimal
+    # perturbation for ord=1.
+    # A common misconcpetion is that FGM for ord=1 consists of dividing
+    # the gradient by its L1 norm.
+    # If you do that for the problem in this unit test, you'll get an
+    # objective function value of ~1.667. The optimal result is 2.
+
+    # We need just one example in the batch and two features to show the
+    # common misconception is suboptimal.
+    grad = tf.placeholder(tf.float32, [1, 2])
+
+    # Build the graph for the attack
+    eta = attacks.optimize_linear(grad, eps=1., ord=1)
+    objective = tf.reduce_sum(grad * eta)
+
+    # Make sure the largest entry of the gradient for the test case is
+    # negative, to catch
+    # the potential bug where we forget to handle the sign of the gradient
+    eta, objective = self.sess.run([eta, objective],
+                                   feed_dict={grad: np.array([[1., -2.]])})
+
+    # Make sure the objective is optimal.
+    # This is the solution obtained by doing the algebra by hand.
+    assert objective == 2., objective
```

Comment:
```
self.assertClose(objective, 2.0)
```

Unit tests should use `self.assert*` methods instead of python assertion. If python assertion fails it will just through AssertException, which will fail unit test, but won't provide very good error messages. 
`self.assert*` methods provide much more intuitive error messages when they fail in the test.

Same applies to asserts below in this method and next method